src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs

```diff
@@ -120,91 +121,67 @@ public override ResultWrapper<IReadOnlyList<SimulateBlockResult>> Execute(
-
-                if (givenTime > lastBlockTime)
+                if (blockToSimulate.BlockOverrides.Time is not null)
                 {
-                    lastBlockTime = givenTime;
+                    if (blockToSimulate.BlockOverrides.Time <= lastBlockTime)
+                    {
+                        return ResultWrapper<IReadOnlyList<SimulateBlockResult>>.Fail(
+                            $"Block timestamp out of order {blockToSimulate.BlockOverrides.Time} is <= than given base timestamp of {lastBlockTime}!", ErrorCodes.BlockTimestampNotIncreased);
+                    }
+                    lastBlockTime = (ulong)blockToSimulate.BlockOverrides.Time;
                 }
                 else
                 {
-                    return ResultWrapper<IReadOnlyList<SimulateBlockResult>>.Fail(
-                        $"Block timestamp out of order {givenTime}!", ErrorCodes.BlockTimestampNotIncreased);
+                    blockToSimulate.BlockOverrides.Time = lastBlockTime + secondsPerSlot;
+                    lastBlockTime = (ulong)blockToSimulate.BlockOverrides.Time;
                 }
-
-                blockToSimulate.BlockOverrides.Time = givenTime;
+                lastBlockNumber = (long)givenNumber;
             }
-
-            long minBlockNumber = Math.Min(
-                call.BlockStateCalls.Min(b => (long)(b.BlockOverrides?.Number ?? ulong.MaxValue)),
-                header.Number + 1);
-
-            long maxBlockNumber = Math.Max(
-                call.BlockStateCalls.Max(b => (long)(b.BlockOverrides?.Number ?? ulong.MinValue)),
-                minBlockNumber);
-
-            HashSet<long> existingBlockNumbers =
-            [
-                .. call.BlockStateCalls.Select(b => (long)(b.BlockOverrides?.Number ?? ulong.MinValue))
-            ];
-
-            List<BlockStateCall<TransactionForRpc>> completeBlockStateCalls = call.BlockStateCalls;
-
-            for (long blockNumber = minBlockNumber; blockNumber <= maxBlockNumber; blockNumber++)
-            {
-                if (!existingBlockNumbers.Contains(blockNumber))
-                {
-                    completeBlockStateCalls.Add(new BlockStateCall<TransactionForRpc>
-                    {
-                        BlockOverrides = new BlockOverride { Number = (ulong)blockNumber },
-                        StateOverrides = null,
-                        Calls = []
-                    });
-                }
-            }
-
-            call.BlockStateCalls.Sort((b1, b2) => b1.BlockOverrides!.Number!.Value.CompareTo(b2.BlockOverrides!.Number!.Value));
+            // sort element of the completeBlockStateCalls by block number
+            completeBlockStateCalls.Sort((a, b) => a.BlockOverrides.Number!.Value.CompareTo(b.BlockOverrides.Number!.Value));
+            call.BlockStateCalls = [.. completeBlockStateCalls];
```

Comment:
Why do we need to sort at the end? Shouldn't we just error out if it wasn't consecutive in the first place?

Reply:
Yes you are correct, there is no need for sorting. Anyway we don't have to check whether the blocks are in consecutive order, as the above the block will error out/

Reply:
Yes, you are correct; we don't need to sort. We don't need to add a check either for the blocks to be consecutive as  the for loop will take care of it